src/adsp_config.c

```diff
@@ -0,0 +1,1158 @@
+/* SPDX-License-Identifier: BSD-3-Clause
```

Comment:
This file needs a lot more inline comments within each function explaining what is going on - difficult to review otherwise.

Reply:
Added much more comments :)